### PR TITLE
Reorganize Windows game launcher to support per-mod launchers.

### PR DESCRIPTION
**Pull request by pchote**
This sets up our windows launcher and packaging to support per-mod launchers for both https://github.com/OpenRA/OpenRAModTemplate and our official mods once we drop the mod chooser.  This is the Windows equivalent of #12824 / https://github.com/OpenRA/OpenRALauncherOSX/pull/2.

https://github.com/pchote/OpenRA/commit/a16576a464b74d1bfcc3cd7e4fde24111d2f47bd shows how this will work for per-mod launchers in a future pr (which is currently blocked by #12909 and some other installer changes).  For now this continues to build a mod-independent OpenRA.exe launcher.

A test installer built using our normal packaging scripts can be found [here](https://www.dropbox.com/s/i7zomvw2su2dbdf/OpenRA-test-20170314.exe?dl=0).
The appveyor test installer can be found [here](https://ci.appveyor.com/project/OpenRA/openra/build/1.0.14695/artifacts).

### CODE REVIEW

**Review by reaperrr — APPROVED** [review]
Looks good to me, and both provided test builds as well as my own compile worked.

File: packaging/windows/WindowsLauncher.cs.in [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+
+		[DllImport("shell32.dll")]
+		static extern IntPtr SHGetFileInfo(string pszPath, uint dwFileAttributes, ref SHFILEINFO psfi, uint cbSizeFileInfo, uint uFlags);
+
+		[DllImport("user32.dll")]
+		public extern static bool DestroyIcon(IntPtr handle);
+
+		struct SHFILEINFO
+		{
+			public IntPtr hIcon; // original type: HICON
```
Comment (abcdefg30) [inline packaging/windows/WindowsLauncher.cs.in]: Comments usually go on their own line, starting with a capital letter. (Same for the two other comments below.)

**Review by abcdefg30 — CHANGES_REQUESTED** [review]
Looks like the removal/moving of `GameMonitor.cs` got lost somewhen during the process.

**Comment by pchote** [comment]
Fixed and rebased.

**Review by abcdefg30 — CHANGES_REQUESTED** [review]
So, with this `make all` doesn't create `OpenRA.exe` and `OpenRA.Game.exe` anymore. Am I missing something here?

**Comment by pchote** [comment]
Not creating OpenRA.exe is intentional:
1.  We have a launch-game.cmd that is more appropriate, including a mod selector instead of requiring a separate exe to be compiled per mod.
2.  We don't build any other platform specific launchers, why should windows be special?
3.  Building the launchers is non-trivial, and i'm strongly against having to maintain multiple different scripts to do this without a very good reason.

This does create `OpenRA.Game.exe`...?  I just tested this again to be certain, using `make.cmd` on a clean checkout.

**Comment by reaperrr** [comment]
> This does create OpenRA.Game.exe...? I just tested this again to be certain, using make.cmd on a clean checkout.

It works here as well.

@abcdefg30 Maybe you overlooked it because it doesn't have the RA icon anymore?

**Review by abcdefg30 — APPROVED** [review]
Weird. I must have messed something up. Did a clean checkout and it works now.

**Pull request merged**